packages/semver/src/executors/version/utils/git.ts

```diff
@@ -86,13 +86,23 @@ export function tryPush({
     );
 }
 
-export function addToStage({
-  paths,
-  dryRun,
-}: {
+export type AddToStage = {
   paths: string[];
   dryRun: boolean;
-}): Observable<void> {
+}
+
+export function mergeAddToStage(stages: AddToStage[], dryRun: boolean): AddToStage {
+  const pathsSet = new Set<string>();
+
+  stages.forEach((stage) => stage.paths.forEach((path) => pathsSet.add(path)));
```

Comment:
I think we don't need an extra function for this, I would put this directly in the `addToStage` function so we make sure paths are always unique when calling `addToStage`.